src/actions/write/evolve.ts

```diff
@@ -1,22 +1,42 @@
-import { DEFAULT_NON_CONTRACT_OWNER_MESSAGE } from '../../constants';
-import { ContractResult, IOState, PstAction } from '../../types';
+import {
+  FOUNDATION_ACTION_PASSED_STATUS,
+  FOUNDATION_DELAYED_EVOLVE_COMPLETED_STATUS,
+} from '../../constants';
+import {
+  ContractResult,
+  DelayedEvolveInput,
+  IOState,
+  PstAction,
+} from '../../types';
 
 declare const ContractError;
+declare const SmartWeave: any;
 
 // Updates this contract to new source code
 export const evolve = async (
   state: IOState,
-  { caller, input: { value } }: PstAction,
+  { input: { value } }: PstAction,
 ): Promise<ContractResult> => {
-  const owner = state.owner;
+  const foundationActions = state.foundation.actions;
```

Comment:
we may want to null coalesce this `state.foundation?.actions ?? {}`

Reply:
Makes sense!